File: Download_pixiv/SendNotify.js
const axios = require('axios');

function sendEnterpriseWechatNotification(message) {
  const corpID = 'YOUR_CORP_ID'; // 企业微信的CorpID
  const agentID = 'YOUR_AGENT_ID'; // 企业微信的应用AgentID
  const chatID = 'YOUR_CHAT_ID'; // 企业微信群聊的ChatID
  const appSecret = 'YOUR_APP_SECRET'; // 企业微信应用的Secret

  const url = `https://qyapi.weixin.qq.com/cgi-bin/appchat/send?access_token=ACCESS_TOKEN`;

  axios.get(`https://qyapi.weixin.qq.com/cgi-bin/gettoken?corpid=${corpID}&corpsecret=${appSecret}`)
  .then(response => {
    const accessToken = response.data.access_token;
    const apiUrl = url.replace('ACCESS_TOKEN', accessToken);

    return axios.post(apiUrl, {
      chatid: chatID,
      msgtype: 'text',
      text: {
        content: message,
      },
    });
  })
  .then(response => {
    console.log('企业微信消息发送成功:', response.data);
  })
  .catch(error => {
    console.error('企业微信消息发送失败:', error);
  });
}

module.exports = {
  sendEnterpriseWechatNotification,
};

File: Download_wallhaven/SendNotify.py
import requests

# 目标域名
target_domain = 'https://qyapi.weixin.qq.com'

# 企业微信应用配置
corpid = 'YOUR_CORP_ID'  # 企业ID
corpsecret = 'YOUR_CORP_SECRET'  # 应用Secret
agentid = 'YOUR_AGENT_ID'  # 应用AgentID

class CustomWeCom:
    def __init__(self, corpid, corpsecret, agentid, target_domain):
        self.CORPID = corpid
        self.CORPSECRET = corpsecret
        self.AGENTID = agentid
        self.target_domain = target_domain

    def get_access_token(self):
        url = f"{self.target_domain}/cgi-bin/gettoken?corpid={self.CORPID}&corpsecret={self.CORPSECRET}"
        response = requests.get(url)
        data = response.json()
        return data.get("access_token")

    def send_text(self, message, touser="@all"):
        access_token = self.get_access_token()
        send_url = f"{self.target_domain}/cgi-bin/message/send?access_token={access_token}"
        send_values = {
            "touser": touser,
            "msgtype": "text",
            "agentid": self.AGENTID,
            "text": {"content": message},
        }
        response = requests.post(send_url, json=send_values)
        return response.json()["errmsg"]

# 推送消息
def send_to_wecom(title, content):
    wx = CustomWeCom(corpid, corpsecret, agentid, target_domain)
    message = f"{title}\n\n{content}"
    response = wx.send_text(message)
    if response == "ok":
        print("企业微信应用推送成功！")
    else:
        print(f"企业微信应用推送失败：{response}")

# 示例用法
if __name__ == "__main__":
    title = "测试标题"
    content = "测试内容"
    send_to_wecom(title, content)
